projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -22,41 +24,85 @@ import styles from './styles.module.scss';
  * @param {boolean} props.isPublicizeEnabled            - Whether Publicize is enabled for this post.
  * @param {boolean} props.isRePublicizeFeatureEnabled   - True if the RePublicize feature is available.
  * @param {boolean} props.isPublicizeDisabledBySitePlan - A combination of the republicize feature being enabled and/or the post not being published.
+ * @param {boolean} props.numberOfSharesRemaining       - The number of shares remaining for the current period. Optional.
  * @returns {object}                                    - Publicize form component.
  */
 export default function PublicizeForm( {
 	isPublicizeEnabled,
 	isRePublicizeFeatureEnabled,
 	isPublicizeDisabledBySitePlan,
+	numberOfSharesRemaining = null,
 } ) {
-	const { connections, toggleById, hasConnections } = useSocialMediaConnections();
+	const {
+		connections,
+		toggleById,
+		hasConnections,
+		enabledConnections,
+	} = useSocialMediaConnections();
 	const { message, updateMessage, maxLength } = useSocialMediaMessage();
 
 	const isDisabled = () =>
 		! isRePublicizeFeatureEnabled && connections.every( connection => ! connection.toggleable );
 	const Wrapper = isPublicizeDisabledBySitePlan ? Disabled : Fragment;
 
+	const outOfConnections =
+		numberOfSharesRemaining !== null && numberOfSharesRemaining <= enabledConnections.length;
+
 	return (
 		<Wrapper>
 			{ hasConnections && (
-				<PanelRow>
-					<ul className={ styles[ 'connections-list' ] }>
-						{ connections.map(
-							( { display_name, enabled, id, service_name, toggleable, profile_picture } ) => (
-								<PublicizeConnection
-									disabled={ isRePublicizeFeatureEnabled ? ! isPublicizeEnabled : ! toggleable }
-									enabled={ enabled && ! isPublicizeDisabledBySitePlan }
-									key={ id }
-									id={ id }
-									label={ display_name }
-									name={ service_name }
-									toggleConnection={ toggleById }
-									profilePicture={ profile_picture }
-								/>
-							)
-						) }
-					</ul>
-				</PanelRow>
+				<>
+					{ numberOfSharesRemaining !== null && (
+						<PanelRow>
+							<Notice type={ numberOfSharesRemaining < connections.length ? 'warning' : 'default' }>
```

Comment:
I know we spoke about this on slack, but I was thinking some more about this. I reckon we should show the notice only when the user doesn't have the jetpack social plan. That's controlled not just by the sticker, but also by the presence of a plan. 

I had a Jetpack Social Basic Monthly plan and could still see the notice. 

So, may be show the notice only if the user doesn't have the paid plan? We can use [this flag ](https://github.com/Automattic/jetpack/blob/fe38d3671f2db213277bd1ea6d1b819b780561a0/projects/plugins/social/src/class-jetpack-social.php#L153:~:text=%24show_nudge%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3D%20!%20Current_Plan%3A%3Asupports(%20%27social%2Dshares%2D1000%27%2C%20%24refresh_plan_from_wpcom%20)%3B) to determine that.

Reply:
Yes, good idea. I'll implement that.

Reply:
Has that been implemented? Is that wrapped up in the `isDisabled` check?

Reply:
It's implemented [here](https://github.com/Automattic/jetpack/pull/25661/files#diff-58f304088488b354751173257592c43671853b5b733dbdc410f057ea0b1c6bf4R88), by only passing in `numberOfSharesRemaining` if the user doesn't have a paid plan.